PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+                        }
+                        self.operationDispatcher.dispatchOnMainThread {
+                            receiveEligibility(result)
+                        }
+                    }
+                }
+            } else {
+                self.backend.getIntroEligibility(appUserID: self.appUserID,
+                                                 receiptData: Data(),
+                                                 productIdentifiers: productIdentifiers) { result, maybeError in
+                    if let error = maybeError {
+                        Logger.error(String(format: "Unable to getIntroEligibilityForAppUserID: %@",
+                                            error.localizedDescription))
+                    }
+                    self.operationDispatcher.dispatchOnMainThread {
+                        receiveEligibility(result)
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Invalidates the cache for purchaser information.
+     *
+     * Most apps will not need to use this method; invalidating the cache can leave your app in an invalid state.
+     * Refer to https://docs.revenuecat.com/docs/purchaserinfo#section-get-user-information for more information on
+     * using the cache properly.
+     *
+     * This is useful for cases where purchaser information might have been updated outside of the app, like if a
+     * promotional subscription is granted through the RevenueCat dashboard.
+     */
+    @objc func invalidatePurchaserInfoCache() {
+        purchaserInfoManager.clearPurchaserInfoCache(forAppUserID: appUserID)
+    }
+
+    #if os(iOS)
+    /**
+     * Displays a sheet that enables users to redeem subscription offer codes that you generated in App Store Connect.
+     */
+    @available(iOS 14.0, *)
+    @objc func presentCodeRedemptionSheet() {
+        storeKitWrapper.presentCodeRedemptionSheet()
+    }
+    #endif
+
+    /**
+     * Use this function to retrieve the `SKPaymentDiscount` for a given `SKProduct`.
+     *
+     * @param discount The `SKProductDiscount` to apply to the product.
+     * @param product The `SKProduct` the user intends to purchase.
+     * @param completion A completion block that is called when the `SKPaymentDiscount` is returned.
+     * If it was not successful, there will be an `NSError`.
+     */
+    @available(iOS 12.2, macOS 10.14.4, macCatalyst 13.0, tvOS 12.2, watchOS 6.2, *)
+    @objc(paymentDiscountForProductDiscount:product:completion:)
+    func paymentDiscount(for discount: SKProductDiscount,
+                         product: SKProduct,
+                         completion: @escaping PaymentDiscountBlock) {
+        purchasesOrchestrator.paymentDiscount(forProductDiscount: discount, product: product, completion: completion)
+    }
+
+    @available(iOS 12.0, macOS 10.14, macCatalyst 13.0, tvOS 12.0, watchOS 6.2, *)
+    private func modernEligibilityHandler(maybeReceiptData maybeData: Data?,
```

Comment:
oh hmm do we need the nil check above is this maybeData is nilable anyway?

Reply:
You're right, I didn't need this to be nil!
✅